jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AllowedResourceAliasChecker.java

```diff
@@ -172,14 +180,20 @@ protected boolean isAllowed(Path path)
             {
                 // If the path is the same file as the base, then it is contained in the base and
                 // is not protected.
-                if (isSameFile(path, _base))
+                if (isSameFile(path, _baseResource))
                     return true;
 
                 // If the path is the same file as any protected resources, then it is protected.
-                for (Path p : _protected)
+                for (String protectedTarget : _protected)
                 {
-                    if (isSameFile(path, p))
-                        return false;
+                    Resource p = _baseResource.resolve(protectedTarget);
```

Comment:
```suggestion
                    Resource r = _baseResource.resolve(protectedTarget);
```

Reply:
Can't do, will conflict with another Resource r.

p for protected resource, seems okay to me.